File: 0027-remove-element/0027-remove-element.cpp
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int left = 0;
        int right = nums.size() - 1;
        
        while (left <= right){
            while (left <= right && nums[right] == val) --right;
            while (left <= right && nums[left] != val) left++;
            if (left < right){
                nums[left] = nums[right];
                left++;
                right--;
            }            
        }
        return left;
    }
};

File: 0027-remove-element/0027-remove-element.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        slow, fast = 0, 0
        while fast < len(nums):
            if nums[fast] != val:
                if nums[slow] == val:
                    nums[slow], nums[fast] = nums[fast], nums[slow]
                slow += 1
            fast += 1
        return slow
        